### PR TITLE
chore: clean up dependency graph repos

### PR DESCRIPTION
**Pull request by hashtagchris**
Update the list of the cli repos, as its been a while. And fix `which`'s association with its repo.

## References

https://github.com/npm/cli/pull/7911/files#r1848890847 - I remembered that `which` has a circular dependency with `@npmcli/eslint-config` if you count devDependencies as well, and noticed `which` wasn't listed.

### CODE REVIEW

File: scripts/npm-cli-repos.txt [hunk 1]
```diff
@@ -54,15 +47,15 @@ promzard
...
 read-package-json-fast
+redact
 rfcs
 run-script
 ssri
 stafftools
 statusboard
-tap-nock
 template-oss
+types
```
Comment (hashtagchris) [inline scripts/npm-cli-repos.txt:58]: I think the `types` repo may have been manually excluded the last time this file was updated, unless the repo was recently tagged `npm-cli`. But leaving it in doesn't change the dependency graph.

**Review by reggi — DISMISSED** [review]

**Comment by wraithgar** [comment]
Gonna have to rebase now that #7911 is merged

**Comment by Yachicken7** [comment]
EQDF6fj6ydJJX_ArwxINjP-0H8zx982W4XgbkKzGvceUWvXl

File: scripts/npm-cli-repos.txt (hunk 1 above)
Comment (wraithgar) [reply scripts/npm-cli-repos.txt:58]: It wasn't tagged before.  It's not in the cli so it won't have an effect outside this file.

**Comment by wraithgar** [comment]
This also removes `fs-minipass` from the cli-repos list.  I think we can squash this w/ the new title I just added.

**Review by wraithgar — APPROVED** [review]

**Pull request merged**